src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java

```diff
@@ -544,15 +547,158 @@ public boolean isPluginEnabled(Plugin plugin)
 	public Plugin loadPlugin(File file)
 			throws InvalidPluginException, InvalidDescriptionException, UnknownDependencyException
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		if (!file.getName().endsWith(".jar")) 
+		{
+			throw new InvalidPluginException("Plugin must be a .jar file");
+		}
+
+		try 
+		{
+			File tempDir = createTemporaryDirectory("MockBukkit-plugin-" + file.getName().replace(".", "-"));
+			Path targetPath = Files.copy(file.toPath(), tempDir.toPath().resolve(file.getName()), StandardCopyOption.REPLACE_EXISTING);
+			Plugin plugin = loader.loadPlugin(targetPath.toFile());
+
+			if (plugin == null) 
+			{
+				throw new InvalidPluginException("Plugin " + file.getName() + "couldn't be loaded.");
+			}
+
+			registerLoadedPlugin(plugin);
+			return plugin;
+		} 
+		catch (IOException ex) 
+		{
+			throw new RuntimeException("Unable to instantiate plugin: ", ex);
+		}
 	}
 
 	@Override
 	public Plugin[] loadPlugins(File directory)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		try 
+		{
+			//copy to temp folder
+			File tempDir = createTemporaryDirectory("MockBukkit-plugins");
+			Path targetPath = Files.copy(directory.toPath(), tempDir.toPath(), StandardCopyOption.REPLACE_EXISTING);
+
+			Map<String, File> pluginsToLoad = new HashMap<>();
+			Map<String, Set<String>> softDependencies = new HashMap<>();
+			Map<String, Set<String>> dependencies = new HashMap<>();
+
+			for(File file : targetPath.toFile().listFiles())
+			{
+				try 
+				{
+					//check description validity
+					PluginDescriptionFile pdf = loader.getPluginDescription(file);
+
+					//acknowledge plugin name restrictions
+					String pluginName = pdf.getName();
+					checkPluginNameValidity(file, pluginName);
+
+					//check if already registered
+					File previousRegisteredPlugin = pluginsToLoad.put(pluginName, file);
+					if (previousRegisteredPlugin != null) 
+					{
+						throw new InvalidDescriptionException(String.format("Plugin %s wasn't loaded: Duplicate plugin name %s!", file.getName(), pluginName));
+					}
+
+					//gather softDependencies, dependencies, and loadBefore
+					Set<String> pluginSoftDependencies = softDependencies.getOrDefault(pluginName, new HashSet<>());
+					pluginSoftDependencies.addAll(pdf.getSoftDepend());
+					softDependencies.put(pluginName, pluginSoftDependencies);
+
+					Set<String> pluginDependencies = dependencies.getOrDefault(pluginName, new HashSet<>());
+					pluginDependencies.addAll(pdf.getDepend());
+					dependencies.put(pluginName, pluginDependencies);
+
+					for (String loadBefore : pdf.getLoadBefore())
+					{
+						Set<String> softDepends = softDependencies.getOrDefault(loadBefore, new HashSet<>());
+						softDepends.add(pluginName);
+						softDependencies.put(loadBefore, softDepends);
+					}
+				}
+				catch(InvalidDescriptionException e)
+				{
+					e.printStackTrace();
```

Comment:
You might want to wrap and rethrow the exception here to make sure that the unit tests will fail properly, instead of just printing a stack trace and continuing. Many test runners hide any output unless the test actually fails.